src/Redux/Missions/MissionSlice.js

```diff
@@ -0,0 +1,60 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+export const getMissions = createAsyncThunk(
+  'Missions/getMissions', // Action
+  async () => {
+    const response = await fetch('https://api.spacexdata.com/v3/missions').then((resp) => resp.json());
+
+    const missionsData = response.map((mission) => {
+      const missions = {
+        mission_id: mission.mission_id,
+        mission_name: mission.mission_name,
+        mission_description: mission.description,
+        reserved: false,
+      };
+      return missions;
+    });
+    return missionsData;
+  },
+);
+
+const initialState = { loading: false, data: [], error: null };
+
+const missionsSlice = createSlice({
+  name: 'missions',
+  initialState,
+  reducers: {
+    joinMission(state, { payload }) {
+      const newMissions = [];
+      state.data.map((mission) => {
+        if (mission.mission_id === payload) {
+          newMissions.push({
+            ...mission,
+            reserved: !mission.reserved,
+          });
+        } else {
+          newMissions.push(mission);
+        }
+        return newMissions;
+      });
+      return { ...state, data: newMissions };
+    },
+  },
```

Comment:
- [ ] The same happens here, you're using the same `joinMission` action to do both: joining and leaving missions. Please consider separating them! remember that using redux, only actions can change the state, so if I click on leave mission and the action `"joinMission"` gets fired, it is not going to make any sense! Please consider applying this change. 🤓 

- [ ] **[OPTIONAL]:** Also, it would be good if you can simplify this logic, one powerful feature of redux toolkit action creators is that you don't have to worry about mutating the state because it uses the library `Immer` under the hood to transform them into immutable changes, so you could simplify it doing something like this:

```js
    joinMission(state, { payload }) {
      state.data.forEach((mission) => {
        if (mission.mission_id === payload) return { ...mission, reserved: true };
        return mission;
      });
    },
}
```

⚠️ WARNING ⚠️ Remember, the "mutating" logic only works correctly when wrapped inside of Immer! Otherwise, that code will really mutate the data. Read more [here](https://redux-toolkit.js.org/usage/immer-reducers).